### PR TITLE
[MHV-43740] Setting focus on h1 tags SM

### PR DESCRIPTION
**Pull request by vsaleem**
## Summary

- *MHV-43740: Updated focus to H1 when user navigates using links*
- *MHV-41708: Text change: Updated 'Compose message' to 'Start a new message' across SM compose screens and unit tests*

- *Va.gov - Secure Messaging*

## Related issue(s)
- department-of-veterans-affairs/vets-website
- *[Link to ticket created in va.gov-team repo: https://vajira.max.gov/browse/MHV-43740*
- *[Link to ticket created in va.gov-team repo: https://vajira.max.gov/browse/MHV-41708*

## Testing done

- *Prior to changes, a screenreader would not announce the page title to the user and 'Compose message' has been updated to 'Start a new message' across all SM screens and links*
- *To verify in desktop view:  Turn on screenreader, navigate to any page using the left navigation menu and press enter, after initial announcements have been read, (Loading secure messages, etc.), the H1 title or next in line heading should be read by the screenreader*
- *Manual testing using the screenreader, axe, and updated cypress tests that looks for focus have been completed*

## Screenshots

| | Before | After |
| --- | --- | --- |
| Mobile |Screenshot 2023-04-25 at 10 28 32 AM|Screenshot 2023-04-25 at 10 24 17 AM|
| Desktop |Screenshot 2023-04-25 at 10 28 47 AM|Screenshot 2023-04-25 at 10 24 04 AM|

## What areas of the site does it impact?
*Va.gov - Secure Messaging*

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [x]  I added a screenshot of the developed feature
- [x]  [Accessibility foundational testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/mhv/secure-messaging/components/MessageDetailBlock.jsx [hunk 1]
```diff
@@ -59,6 +60,10 @@ const MessageDetailBlock = props => {
     [location.pathname, dispatch],
   );
 
+  useEffect(() => {
+    focusElement(document.querySelector('h2'));
+  });
+
```
Comment (oleksii-morgun-va) [inline src/applications/mhv/secure-messaging/components/MessageDetailBlock.jsx:66]: I am curious if this h2 header for message subject should actually be an h1 instead? I am referring to  the element itself, rather then the focus reference. Just thinking that in this case, this page would not have a valid h1 then if this message subject remains h2

File: src/applications/mhv/secure-messaging/containers/AuthorizedRoutes.jsx [hunk 2]
```diff
@@ -13,7 +13,7 @@ import SearchResults from './SearchResults';
 
 const AuthorizedRoutes = () => {
   return (
-    <div className="vads-u-flex--fill">
+    <div className="secure-messaging vads-u-flex--fill">
```
Comment (oleksii-morgun-va) [inline src/applications/mhv/secure-messaging/containers/AuthorizedRoutes.jsx:16]: Let's double check if this is really needed here. Maybe I missed, but I don't see anywhere where we would be relying on this class. Additionally, we are already declaring a `secure-messaging-container` class in App.jsx. However, if there is a plan for this class to be used, we can keep it

File: src/applications/mhv/secure-messaging/containers/FolderListView.jsx [hunk 3]
```diff
@@ -132,7 +132,7 @@ const FolderListView = props => {
 
   return (
     <div className="vads-l-grid-container vads-u-padding--0">
-      <div className="main-content">
+      <div className="main-content folder-list-view">
```
Comment (oleksii-morgun-va) [inline src/applications/mhv/secure-messaging/containers/FolderListView.jsx:135]: same as above for `secure-messaging` class. We could always use `data-testid `attribute as well if we are not relying on this class declaration for styling

File: src/applications/mhv/secure-messaging/tests/containers/FolderListView.unit.spec.jsx [hunk 4]
```diff
@@ -1,62 +1,70 @@
...
-import messages from '../fixtures/search-response.json';
-import folder from '../fixtures/folder-inbox-metadata.json';
-import reducer from '../../reducers';
-import FolderListView from '../../containers/FolderListView';
+/**
+ * @Note : This component is currently not used in SM application.
+ *         Consider deleting this file if there are no plans to use
+ *         this component in the future, or refactor this file to test
+ *         for the 'FilterBox' component.
+ */
```
Comment (oleksii-morgun-va) [inline src/applications/mhv/secure-messaging/tests/containers/FolderListView.unit.spec.jsx:6]: I agree, we can get rid of FolderListView component and this unit test. @robertbylight should have had test coverage done for FilterBox on his branch which is yet to be merged.

**Review by alex-morgun-1 — CHANGES_REQUESTED** [review]

File: src/applications/mhv/secure-messaging/containers/AuthorizedRoutes.jsx (hunk 2 above)
Comment (vsaleem) [reply src/applications/mhv/secure-messaging/containers/AuthorizedRoutes.jsx:16]: We're not using this classname officially for styling yet but it did help with my debugging by identifying the sections. For that purpose I'll change it to a data-testid. Thanks!

**Review by alex-morgun-1 — APPROVED** [review]

**Pull request merged**